File: 백준/Silver/1912. 연속합/연속합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int max = arr[0];
        int cur = arr[0];

        for (int i = 1; i < n; i++) {
            cur = Math.max(cur + arr[i], arr[i]);
            max = Math.max(max, cur);
        }

        System.out.print(max);
    }
}

File: 백준/Silver/1912. 연속합/연속합.kt
import java.lang.Integer.max

fun main() = with(System.`in`.bufferedReader()) {
    val n = readLine().toInt()
    val number = readLine().split(' ').map { it.toInt() }
    val dp = IntArray(n)
    dp[0] = number[0]
    var max = number[0]
    for (i in 1 until n) {
        dp[i] = max(dp[i - 1] + number[i], number[i])
        max = max(max, dp[i])
    }
    print(max)
}